Packs/IdentityRecordedFuture/Integrations/IdentityRecordedFuture/IdentityRecordedFuture.py

```diff
@@ -0,0 +1,437 @@
+
+        if isinstance(identities, str):
+            identities = identities.replace(" ", "").split(";")
+            for identity in identities:
+                if re.fullmatch(emailRegex, identity):
+                    email_identities.append(identity)
+                # If user used sha1 value for lookup try to find it in authorization data and email data
+                elif re.fullmatch(sha1Regex, identity):
+                    sha1_identities.append(identity)
+                    for domain in domains:
+                        authorization_identities.append(
+                            {
+                                "login_sha1": identity,
+                                "domain": domain,
+                            }
+                        )
+                else:
+                    for domain in domains:
+                        authorization_identities.append(
+                            {
+                                "login": identity,
+                                "domain": domain,
+                            }
+                        )
+        elif isinstance(identities, dict):
+            authorization_identities.append(identities)
+        else:
+            raise DemistoException(
+                f"Failed due to - Received unexpected data in Lookup command: {identities}"
+            )
+
+        date_period = period_to_date(first_downloaded)
+        response = self.client.identity_lookup(
+            email_identities,
+            authorization_identities,
+            sha1_identities,
+            date_period,
+            password_properties,
+        )
+        identities_data = response.get("identities", [])
+        command_results = self.__build_lookup_context(
+            identities_data, identities
+        )
+        return command_results
+
+    def __build_lookup_markdown(self, lookup_data, identities):
+        if lookup_data:
+            markdown = ["## Credentials Lookup"]
+            markdown.append("*****")
+            for identity in lookup_data:
+                markdown.append(
+                    f'### Identity __{identity["identity"]["subjects"][0]}__:'
+                )
+                markdown.append("*****")
+                credentials = identity["credentials"]
+                passwords_section = []
+                dumps_section = []
+                breaches_section = []
+                for idx, credential in enumerate(credentials):
+                    exposed_secret = credential["exposed_secret"]
+                    password_number_text = f"__Password {idx + 1}__"
+                    passwords_section.append(f"{password_number_text}:")
+                    if exposed_secret.get("details", {}).get("rank"):
+                        passwords_section.append(
+                            f"__Rank__: {exposed_secret['details']['rank']}"
+                        )
+                    if exposed_secret[
+                        "effectively_clear"
```

Comment:
is `effectively_clear` always a key of `exposed_secret`? 

if not, it's recommended to use `exposed_secret.get("effectively_clear",<True or False>)` instead.

Reply:
It is. But just to protect the code I will use `get`